registry/client/repository.go

```diff
@@ -99,10 +98,6 @@ func (r *registry) Repositories(ctx context.Context, entries []string, last stri
 
 // NewRepository creates a new Repository for the given repository name and base URL.
 func NewRepository(ctx context.Context, name, baseURL string, transport http.RoundTripper) (distribution.Repository, error) {
-	if _, err := reference.ParseNamed(name); err != nil {
-		return nil, err
-	}
-
```

Comment:
I think the purpose of this code is to validate the name passed in. It's not really the right way to do it - `NewRepository` should accept a `reference.Named` instead of a string for the name - but I don't want to slow down this PR by making unrelated changes. For now, could you please leave this validation in place? I'll open a separate PR to change the argument type.